2009-flu-uk.qmd

```diff
@@ -1 +1,437 @@
+
+```{r}
+packer <- monty_packer(c("alpha", "R_0", "rho", "eta"),
+                       fixed = list(h_times = hol_t, h_values= hol_v))
+```
+
+This defines two functions that allows to 'pack' (define) and 'unpack' (define) our parameters.
+
+Now building the `monty` model for the likelihood of the model is very easy:
+
+```{r}
+likelihood <- dust_likelihood_monty(filter, packer)
+```
+
+### Setting up the prior distribution
+
+We similarly set up a log_prior function. Note that we use the function to exclude impossible values such as proportion below 0 or above 1. This is to avoid to get logdensity of -infinity given that these values would have a density value of 0 (=impossible).
+
+The other ingredient we need is a prior. This we can construct with monty_dsl as before, note that the names are matching the parameters from the likelihood as it should be.
+
+```{r}
+prior <- monty_dsl({
+  alpha ~ Beta(a = 4e-4, b = 2)
+  R_0 ~ Gamma(2, 0.7)
+  rho ~ Uniform(0, 1)
+  eta ~ Exponential(mean = 1000)
+})
+```
+
+Let's inspect the our prior `monty` model
+```{r}
+prior
+```
+
+As it has been built using the monty DSL, it comes with gradient calculation, direct sampling (so no need to use one of the Monte Carlo algorithm) and accepts multiple parameters.
+
+Let's try to generate samples and check that they match our prior distribution:
+```{r}
+n_streams <- 10000
+r <- monty_rng$new(n_streams = n_streams)
+prior_samples <- matrix(monty_model_direct_sample(prior, r), nrow = n_streams)
+colnames(prior_samples) <- prior$parameters
+```
+
+```{r}
+bayesplot::mcmc_pairs(prior_samples)
+```
+
+### Setting up the posterior
+
+Our posterior is the product of the likelihood and prior, or the sum of their logs:
+
+```{r}
+posterior <- likelihood + prior
+```
+
+### Chosing the sampler
+
+Next, we define a sampler; we'll start with a random walk with a fairly arbitrary diagonal proposal matrix:
+
+```{r}
+vcv <- diag(c(5e-10,5e-5,1e-5,1))
+sampler <- monty_sampler_adaptive(vcv, initial_vcv_weight = 100)
+```
+
+We start this off, using explicit initial conditions
+
+```{r}
+#| cache: true
+init_values <- packer$pack(list(alpha=par0[1],
+                  R_0=par0[2],
+                  rho=0.1,
+                  eta=10,
+                  h_times = hol_t,
+                  h_values= hol_v))
+init_values <- c(3.05e-5,1.354,0.04, 10)
+samples <- monty_sample(posterior, sampler, 10000, initial = init_values,
+                        n_chains = 3)
+```
+
+::: {.callout-note}
+We have used explicit initial conditions here, which might not be what you want in all situations.  Better might be to sample from the prior, but we have not yet implemented support to try a few points from the sample before getting a point with finite density, which is really needed here.
```

Comment:
```suggestion
We have used explicit initial conditions here, which might not be what you want in all situations.  It might be better to sample from the prior, but we have not yet implemented support to try a few points from the sample before getting a point with finite density, which is really needed here.
```

Reply:
This is also something we need to fix, but it's actually quite annoying to get right - I've had a look a couple of times. In the other cases I've seen though, if the observation model is good enough and parameters appropriately bounded (incl by the prior) then you can usually omit the initial conditions - might be worth a try here if you're assuming it will fail